pkg/internal/testing/addr/manager.go

```diff
@@ -42,16 +42,30 @@ var (
 )
 
 func init() {
-	baseDir, err := os.UserCacheDir()
+	var err error
+	cacheDir, err = tryUserCacheDir()
 	if err != nil {
-		baseDir = os.TempDir()
+		// Either we didn't get a cache directory, or we can't use it
+		cacheDir, err = tryCacheInDir(os.TempDir())
 	}
-	cacheDir = filepath.Join(baseDir, "kubebuilder-envtest")
-	if err := os.MkdirAll(cacheDir, 0750); err != nil {
+	if err != nil {
 		panic(err)
```

Comment:
Instead of moving everything to two functions, can we just add the fallback here?

Reply:
We can, it ends up looking like

```go
func init() {
	baseDir, err := os.UserCacheDir()
	if err == nil {
		cacheDir = filepath.Join(baseDir, "kubebuilder-envtest")
		err = os.MkdirAll(cacheDir, 0o750)
	}
	if err != nil {
		// Either we didn't get a cache directory, or we can't use it
		baseDir = os.TempDir()
		cacheDir = filepath.Join(baseDir, "kubebuilder-envtest")
		err = os.MkdirAll(cacheDir, 0o750)
	}
	if err != nil {
		panic(err)
	}
}
```

The first `err` check isn’t idiomatic and might trip up people making changes here in future, which is why I split everything up, but if that looks OK to you I’ll update the PR accordingly.

Reply:
@vincepri which do you prefer? The version in the PR, or the version in the comment above?

Reply:
The one above works

Reply:
Done, thanks!